pydepsi/network.py

```diff
@@ -0,0 +1,178 @@
+"""Module for creating networks from STM points."""
+
+import math
+
+import numpy as np
+from scipy.spatial import Delaunay
+
+
+def get_distance(s, t):
+    """Calculate the distance between two points.
+
+    Args:
+    ----
+        s: the source point.
+        t: the target point.
+
+    Returns:
+    -------
+        The distance between the two points.
+    """
+    return math.dist(s, t)
+
+
+def check_in_range(value, min_length, max_length):
+    """Check whether a value is inside a specified range.
+
+    Args:
+    ----
+        value: float, value to check.
+        min_length: float, minimum allowed value, inclusive.
+        max_length: float, maximum allowed value, inclusive.
+
+    Returns:
+    -------
+        boolean: true if and only if the value is inside [min_length, max_length].
+    """
+    return min_length <= value and (max_length is None or value <= max_length)
+
+
+def generate_arcs(
+    stm_points, method="delaunay", x="lon", y="lat", min_length=0.00001, max_length=None, max_links=12, num_groups=8
+):
+    """Generate a network from a list of STM points.
+
+    The network is undirected and without self-loops.
+
+    Args:
+    ----
+        stm_points: Xarray.Dataset, input Space-Time Matrix.
+        method: method to form the network; either "delaunay" or "redundant".
+        x: str, first coordinate used to describe a point.
+        y: str, second coordinate used to describe a point.
+        min_length: float, minimum length of any generated arc.
+        max_length: float, maximum length of any generated arc or None.
+        max_links: int, maximum number of arcs per node for the redundant method.
+        num_groups: int, number of parts to split the orientations around each node into for the redundant method.
+
+    Returns:
+    -------
+        arcs: list of pairs, point indices describing the adjacent nodes. The pairs are sorted, as is the list.
+    """
+    if min_length <= 0:
+        print(f"min_length must be positive (currently: {min_length})")
```

Comment:
I replaced the print messages by logger messages.

However, testing shows that info messages do not show in the example notebook when an invalid argument is provided. Instead the code will fail because of incorrect return values without any message with the reason.

When using logger.warning, the example notebook displays these messages as desired.
I set them to 'warning' instead of 'info'. Let me know if I should make them 'info' regardless.

Reply:
Hi Thijs, I think we shoud still use info, since the information here are not warning messages.

By default, Jupyter kernel uses `warning` as default level. I guess that's why the `info` does not should up. Maybe try the following setting in your Notebook? I think they will enable displaying the `INFO` messages.

```python
import logging
logging.basicConfig(level=logging.INFO)
```

Reply:
@vanlankveldthijs sorry given another look in the context, I realized maybe the information here should be `error`?maybe let update that?